var/spack/repos/builtin/packages/py-onnx-runtime/package.py

```diff
@@ -0,0 +1,121 @@
+    patch('libiconv.patch', level=0, when='@1.7.2')
+    # https://github.com/microsoft/onnxruntime/commit/de4089f8cbe0baffe56a363cc3a41595cc8f0809.patch
+    patch('gcc11.patch', level=1, when='@1.7.2')
+
+    dynamic_cpu_arch_values = ('NOAVX', 'AVX', 'AVX2', 'AVX512')
+
+    variant('dynamic_cpu_arch', default='AVX512',
+            values=dynamic_cpu_arch_values, multi=False,
+            description='AVX support level')
+
+    generator = 'Ninja'
+    root_cmakelists_dir = 'cmake'
+
+    def setup_build_environment(self, env):
+        value = self.spec.variants['dynamic_cpu_arch'].value
+        value = self.dynamic_cpu_arch_values.index(value)
+        env.set('MLAS_DYNAMIC_CPU_ARCH', str(value))
+
+    def setup_run_environment(self, env):
+        value = self.spec.variants['dynamic_cpu_arch'].value
+        value = self.dynamic_cpu_arch_values.index(value)
+        env.set('MLAS_DYNAMIC_CPU_ARCH', str(value))
+
+    def cmake_args(self):
+        define = self.define
+        define_from_variant = self.define_from_variant
+
+        args = [define('onnxruntime_ENABLE_PYTHON', True),
+                define('onnxruntime_BUILD_SHARED_LIB', True),
+                define_from_variant('onnxruntime_USE_CUDA', 'cuda'),
+                define('onnxruntime_BUILD_CSHARP', False),
+                define('onnxruntime_USE_EIGEN_FOR_BLAS', True),
+                define('onnxruntime_USE_OPENBLAS', False),
+                define("onnxruntime_USE_MKLML", False),
+                define("onnxruntime_USE_NGRAPH", False),
+                define("onnxruntime_USE_OPENMP", False),
+                define("onnxruntime_USE_TVM", False),
+                define("onnxruntime_USE_LLVM", False),
+                define("onnxruntime_ENABLE_MICROSOFT_INTERNAL", False),
+                define("onnxruntime_USE_BRAINSLICE", False),
+                define("onnxruntime_USE_NUPHAR", False),
+                define("onnxruntime_USE_TENSORRT", False),
+                define("onnxruntime_CROSS_COMPILING", False),
+                define("onnxruntime_USE_FULL_PROTOBUF", True),
+                define("onnxruntime_DISABLE_CONTRIB_OPS", False),
+                define("onnxruntime_USE_PREINSTALLED_PROTOBUF", True),
+                define("onnxruntime_PREFER_SYSTEM_LIB", True)]
+
+        if self.spec.satisfies('+cuda'):
+            args.extend((
+                define('onnxruntime_CUDA_VERSION', str(self.spec['cuda'].version)),
+                define('onnxruntime_CUDA_HOME', self.spec['cuda'].prefix),
+                define('onnxruntime_CUDNN_HOME', self.spec['cudnn'].prefix),
+                define('CMAKE_CUDA_FLAGS', '-cudart shared'),
+                define('CMAKE_CUDA_RUNTIME_LIBRARY', 'Shared'),
+                define('DCMAKE_TRY_COMPILE_PLATFORM_VARIABLES',
+                       'CMAKE_CUDA_RUNTIME_LIBRARY')
+            ))
+
+        return args
+
+    def setup_file(self):
+        return join_path(self.stage.source_path, 'setup.py')
```

Comment:
Can this be removed?

Reply:
Not yet - otherwise Spack won't find it.